bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obsrecord/VisitTimes.java

```diff
@@ -57,4 +72,41 @@ ObsTimeCharges getTimeCharges(ChargeClass mainChargeClass) {
         charges[index] = charges[index].addTime(_unclassifiedTime);
         return new ObsTimeCharges(charges);
     }
+
+    public long getTotalTime() {
+        long result = _unclassifiedTime;
+        for (long t : _classifiedTimes) result += t;
+        return result;
+    }
+
+    /**
+     * Gets the amount of time that is charged to the program or partner (or
+     * unclassified which is eventually charged to one or the other).
+     */
+    public long getChargedTime() {
+        return getTotalTime() - getClassifiedTime(ChargeClass.NONCHARGED);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(_unclassifiedTime, _classifiedTimes);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I added `equals` and `toString` for testing and debugging.  `hashCode` came along for the ride.